src/math/rect.ts

```diff
@@ -1,21 +1,60 @@
 import Vector2 from './vector2.ts';
+import Line from './line.ts';
+import { AABB, ShapeInterface } from "./shape.ts";
+import * as geometry from "./geometry-2d.ts";
 
-export default class Rect {
-    x: number = 0;
-    y: number = 0;
-    width: number = 0;
-    height: number = 0;
+export default class Rect implements ShapeInterface<Rect>{
+    type : "Rect" = "Rect"
+    aabb: AABB;
+    position : Vector2 = Vector2.ZERO;
+    size : Vector2 = Vector2.ZERO;
```

Comment:
```suggestion
    position: Vector2 = Vector2.ZERO;
    size: Vector2 = Vector2.ZERO;
```
I need to fix the linter. :/